tests/tape/tapes/test_qnode.py

```diff
@@ -600,3 +601,31 @@ def func(x, y):
 
         assert np.allclose(res, res2, atol=tol, rtol=0)
         assert func.qtape is not old_tape
+
+
+class TestQNodeCollection:
+    """Unittests for the QNodeCollection"""
+
+    def test_multi_thread(self):
+        n_qubits = 4
+        n_batches = 5
+        dev = qml.device("default.qubit", wires=n_qubits)
+
+        qml.enable_tape()
+
+        def circuit(inputs, weights):
+            qml.templates.AngleEmbedding(inputs, wires=range(n_qubits))
+            qml.templates.BasicEntanglerLayers(weights, wires=range(n_qubits))
+            return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_qubits)]

```

Comment:
We are currently in the process of making sure that templates are compatible with tape more (see #873), so these particular templates might not currently work!

Reply:
Thanks for the review! I will update using basic gates.

Reply:
Maybe this has already been updated, but `AngleEmbedding` should be ok?